backend/de.metas.business/src/main/java/de/metas/order/paymentschedule/InvoicePayScheduleService.java

```diff
@@ -50,9 +60,48 @@ public void createInvoicePaySchedules(final I_C_Invoice invoice)
 
 	public void deleteByInvoiceId(@NonNull final InvoiceId invoiceId) {invoicePayScheduleRepository.deleteByInvoiceId(invoiceId);}
 
-	public void create(@NonNull final InvoicePayScheduleCreateRequest request)
+	public void create(@NonNull final InvoicePayScheduleCreateRequest request) {invoicePayScheduleRepository.create(request);}
+
+	private Optional<InvoicePaySchedule> getByInvoiceId(@NonNull final InvoiceId invoiceId) {return invoicePayScheduleRepository.getByInvoiceId(invoiceId);}
+
+	public boolean validate(@NonNull final InvoiceId invoiceId)
+	{
+		final AtomicBoolean isValid = new AtomicBoolean(false);
+
+		final org.compiere.model.I_C_Invoice invoiceRecord = invoiceBL.getById(invoiceId);
+		getByInvoiceId(InvoiceId.ofRepoId(invoiceRecord.getC_Invoice_ID()))
+				.ifPresent(invoicePaySchedule -> {
+
+					final ImmutableList<InvoicePayScheduleLine> schedules = invoicePaySchedule.getLines();
+					if (schedules.isEmpty())
+					{
+						invoiceRecord.setIsPayScheduleValid(false);
+						invoiceBL.save(invoiceRecord);
+						isValid.set(false);
+					}
+
+					final BigDecimal totalDue = schedules.stream()
```

Comment:
this coudl be a method in InvoicePaySchedule
and it could return Optional<Money>